app/models/user.rb

```diff
@@ -61,4 +61,96 @@ def generate_token(column)
       break unless User.exists?(column => self[column])
     end
   end
+
+  # @return [TrueClass, FalseClass]
+  def uploaded_before? = !user_resources.all.empty?
+
+  # noinspection RubyNilAnalysis
+  def last_uploaded_file
+    return 'none' unless uploaded_before?
+
+    user_resources.last.name
+  end
+
+  # @return Array[Hash]
+  def upload_file_history
+    upload_file_history = []
+    all_upload_dates.each do |item|
+      upload_file_history.append({ 'date': item, 'number': get_number_of_files_at_date(item) })
+    end
+    upload_file_history
+  end
+
+  # @return Array[Hash]
+  # rubocop:disable Metrics/MethodLength
+  def number_of_files_per_type
+    hash_array = []
+    types = {}
+    user_resources.each do |item|
+      if types.include? item.type.to_s
+        types[item.type.to_s] += 1
+      else
+        types[item.type.to_s] = 1
+      end
+    end
+
+    types.each do |item|
+      hash_array.append({ 'type': item[0], 'number': item[1] })
+    end
+    hash_array
+  end
+
+  # rubocop:enable Metrics/MethodLength
+
+  # @return Array[Hash]
+  def top_five_files_by_size
+    hash_array = []
+    user_resources.each do |item|
+      hash_array.append({ 'file': item.name, 'size': item.byte_filesize })
+    end
+    hash_array.sort! { |a, b| a[:size] <=> b[:size] }
+    if hash_array.size >= 5
+      hash_array.reverse.first 5
+    else
+      hash_array.reverse
+    end
+  end
+
+  def times_of_login
+    login_counter
+  end
+
+  def total_number_of_uploads
+    user_resources.count
+  end
+
+  def total_upload_size
+    total_upload_size = 0
+    user_resources.each do |item|
+      total_upload_size += item.byte_filesize
+    end
+    total_upload_size
```

Comment:
Ähm nein, dafür das Framework und DB benutzen, die wesentlich effizienter sind!

`user_resources.sum("byte_filesize")`

https://edgeguides.rubyonrails.org/active_record_querying.html#sum

Reply:
funktioniert leider nicht so einfach
ich finde leider gerade nicht heraus, wie man direkt auf die active_storage_blobs datenbank zugreifen kann, haben sie da eine idee?

Reply:
Na du hast doch Active Storage Setup gemacht, oder nicht? Da hast du doch bestimmt die Migration bemerkt?

Findest du unter den Migrationen und im `db/schema.rb` File.
https://github.com/Ktechen/Dripbox/blob/main/db/migrate/20210625124241_create_active_storage_tables.active_storage.rb

Das sind 1x1 AR-Basics :)

Reply:
die migration ist mir schon klar ^^ die referenz habe ich ja schon in meinem user_resource model angegeben